components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/openjdk/nashorn/JsOpenJdkNashornGraphBuilder.java

```diff
@@ -1221,7 +1226,18 @@ public Object evaluate(AuthenticationContext authenticationContext, Object... pa
                     }
 
                 } catch (Throwable e) {
-                    //We need to catch all the javascript errors here, then log and handle.
+                    // We need to catch all the javascript errors here, then log and handle.
+                    if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                        DiagnosticLog.DiagnosticLogBuilder diagnosticLogBuilder = new DiagnosticLog
+                                .DiagnosticLogBuilder(FrameworkConstants.LogConstants.AUTHENTICATION_FRAMEWORK,
+                                EXECUTE_ADAPTIVE_SCRIPT);
+                        diagnosticLogBuilder.resultMessage("Error in executing the adaptive authentication script : " +
+                                        e.getMessage())
+                                .inputParam(APPLICATION_NAME, authenticationContext.getServiceProviderName())
```

Comment:
Any reasons for not using AppID here? AppName can change

Reply:
This will be fixed with https://github.com/wso2/carbon-identity-framework/pull/4809